Neos.Flow/Classes/Mvc/Routing/Router.php

```diff
@@ -267,9 +267,70 @@ protected function createRoutesFromConfiguration()
             }
             $this->routes[] = $route;
         }
+
         $this->routesCreated = true;
     }
 
+    /**
+     * Emits a signal when the router configuration has been loaded
+     *
+     * This signal can be used to add routes during runtime. In the slot make sure to receive the
+     * $routerConfiguration array by reference so you can alter it.
+     *
+     * @param array $routerConfiguration The router configuration
+     * @return void
+     * @Flow\Signal
+     */
+    protected function emitConfigurationLoaded(array &$routerConfiguration): void
+    {
+    }
+
+    public function configureRouteAnnotatedMethods(array &$routerConfiguration): void
+    {
+        $annotatedRoutes = static::resolveRouteAnnotatedMethods($this->objectManager);
+        foreach ($annotatedRoutes as $annotatedRoute) {
+            /**
+             * Split controller class to values for the '@{package|subpackage|controller|action}' values
+             * Add to routes configuration
+             * createRoutesFromConfiguration will take care of validating the configuration (mismatch, httpmethods and uri used before, etc)
```

Comment:
See `ActionRequest::setControllerObjectName` for a start. Maybe something can be extracted, maybe something has to be c&p